jamjar/jamjar/tasks/transcode_video.py

```diff
@@ -95,75 +100,61 @@ def delete_source(self):
         video_dir = self.video.get_video_dir()
         shutil.rmtree(video_dir)
 
+    def update_jamstarts(self):
+        concert = self.video.concert
+
+        relevant_edges_query = Q(confidence__gt=settings.CONFIDENCE_THRESHOLD) & Q(video1__concert_id=concert.id) & Q(video2__concert_id=concert.id)
+        relevant_edges = Edge.objects.filter(relevant_edges_query)
+
+        # unique the resulting edges and make all edges positive
+        digraph_edges = []
+        seen_video_ids = set()
+        for edge in relevant_edges:
+            video_ids = (edge.video1_id, edge.video2_id)
+
+            if video_ids in seen_video_ids or video_ids[0] == video_ids[1]:
+                continue
+            else:
+                seen_video_ids.add(video_ids)
```

Comment:
the `relevant_edges_query` returns duplicated edges from the db because of the query self-joining. I tried to get it to return DISTINCT video1_id, video2_id records, but didn't have much luck with doing it through the ORM.

To make the list unique, i stored the tuple of (video1_id, video2_id) in the set, so that if (video1_id, video2_id) is seen again, it is ignored